paasta_tools/deployd/common.py

```diff
@@ -62,3 +69,27 @@ def rate_limit_instances(instances, number_per_minute, watcher_name):
 def exponential_back_off(failures, factor, base, max_time):
     seconds = factor * base ** failures
     return seconds if seconds < max_time else max_time
+
+
+def get_service_instances_with_changed_id(marathon_client, instances, cluster):
```

Comment:
This is already in deployd, just moving it around. The oracle is slightly different because it is also looking at number of at risk tasks iirc. That said, this doesn't find apps which have both new and old apps in Marathon and I think it probably should...

I'll have a bit more of a think before we merge this